route/pool.go

```diff
@@ -473,6 +484,25 @@ func (p *EndpointPool) MarshalJSON() ([]byte, error) {
 	return json.Marshal(endpoints)
 }
 
+// OverrulePoolLoadBalancingAlgorithm Overwrites the load balancing algorithm of a pool by that of a specified endpoint, if that is valid.
+func (p *EndpointPool) OverrulePoolLoadBalancingAlgorithm(endpoint *Endpoint) {
```

Comment:
Since you only require the `Endpoint.LoadBalancingAlgorithm` field I would only pass in that string and rename the function, maybe `SetLoadBalancingAlgorithm`.

Reply:
done